sqlglot/dialects/tsql.py

```diff
@@ -219,6 +220,50 @@ def _parse_convert(self, strict):
             # Entails a simple cast without any format requirement
             return self.expression(exp.Cast if strict else exp.TryCast, this=this, to=to)
 
+        def _parse_term(self):
```

Comment:
take a look at sqlglot/transforms and how that is used. something like this would go there

Reply:
Ill have a look thanks!